wdl-analysis/src/scope.rs

```diff
@@ -362,8 +382,12 @@ impl<'a> ScopeRefMut<'a> {
 struct Task {
     /// The span of the task name.
     name_span: Span,
-    /// The scope index for the task.
+    /// The root scope index for the task.
     scope: ScopeIndex,
+    /// The scope index for the outputs.
+    outputs: Option<ScopeIndex>,
+    /// The scope index for the command.
+    command: Option<ScopeIndex>,
```

Comment:
If the command block is required, why is this behind an `Option<>`?

Reply:
That's a good question.

While the command block is required and we emit a diagnostic for it during validation, analysis is designed in such a way as to still supports analyzing on invalid documents to emit additional diagnostics.

So it's written in places to assume that if we haven't seen something because it's invalid, validation emitted a diagnostic for it and otherwise ignore it.

Reply:
Makes sense! Carry on then 👍